docs/src/operate/security/certificate-signing-request.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Certificate signing request
+tags: [Operate, Security, Cloud]
+description: Generate certificate signing request for %%te%%
+---
+
+If you want to use a device certificate which is signed by a Certificate Authority (CA), you can generate the Certificate Signing Request (CSR), which is later used by CA to generate a device certificate. This process requires additional tooling as %%te%% provides you only with the CSR. 
+
+## Create a certificate signing request
+
+To create a CSR you can use [`tedge cert create-csr`](../../references/cli/tedge-cert.md) %%te%% command:
+
+```sh
+sudo tedge cert create-csr --device-id alpha
+```
+
+or
+
+```sh
+sudo tedge cert create-csr
+```
+if a self-signed certificate already exists and you want to reuse device name.
+
+
+```text title="Output"
+Certificate Signing Request was successfully created.
+```
+
+:::note
+`tedge cert` requires `sudo` privilege as creating a private key owned by the MQTT broker. This command provides no output on success.
```

Comment:
Though the command only creates a private key if one does not already exist right?

Reply:
Actually, we can leave it as is, as you need sudo access as well when creating the csr in the default location (under `/etc/tedge/device-certs`).

Reply:
Things are subtle here - but I don't think this needs to be documented.
- root privileges are required for `sudo tedge cert create-cs --device-id xyz` because a new private key is generated
- no privilege is required for `sudo tedge cert create-cs` because the private key is reused.

I think a better approach would have been to `chown` no files inside the command, and to ask the user to run `sudo -u mosquitto tedge cert create  --device-id xyz`  using the appropriate owner for the certificate.